### PR TITLE
Support Pruning new API for TF backend

### PR DESCRIPTION
**Pull request by zehao-intel**
Signed-off-by: zehao-intel <zehao.huang@intel.com>

## Type of Change

feature or bug fix or documentation or validation or others  
API changed or not

## Description

detail description 
JIRA ticket: 
[2672](https://jira.devtools.intel.com/browse/ILITV-2672)
[2616](https://jira.devtools.intel.com/browse/ILITV-2616)

## Expected Behavior & Potential Risk

the expected behavior that triggered by this PR : As title

## How has this PR been tested?

how to reproduce the test (including hardware information): preci, extension

## Dependency Change?

any library dependency introduced or removed: No

### CODE REVIEW

**Comment by chensuyue** [comment]
please fix the link: https://github.com/intel/neural-compressor/tree/zehao/prune/examples#pruning

**Comment by zehao-intel** [comment]
> please fix the link: https://github.com/intel/neural-compressor/tree/zehao/prune/examples#pruning

The links has been updated.

File: examples/tensorflow/image_recognition/inception_v3/pruning/magnitude/README.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+TensorFlow [models](https://github.com/tensorflow/models) repo provides [scripts and instructions](https://github.com/tensorflow/models/tree/master/research/slim#an-automated-script-for-processing-imagenet-data) to download, process and convert the ImageNet dataset to the TF records format.
+We also prepared related scripts in `imagenet_prepare` directory. To download the raw images, the user must create an account with image-net.org. If you have downloaded the raw data and preprocessed the validation data by moving the images into the appropriate sub-directory based on the label (synset) of the image. we can use below command ro convert it to tf records format.
+
+```shell
+cd examples/tensorflow/image_recognition/inception_v3/
+# convert validation subset
+bash prepare_dataset.sh --output_dir==/pruning/magnitude/data/  --raw_dir=/PATH/TO/img_raw/val/ --subset=validation
+# convert train subset
+bash prepare_dataset.sh --output_dir=/pruning/magnitude/data/ --raw_dir=/PATH/TO/img_raw/train/ --subset=train
+cd /pruning/magnitude/
```
Comment (lvliang-intel) [inline examples/tensorflow/image_recognition/inception_v3/pruning/magnitude/README.md:30]: cd ./pruning/magnitude/

File: examples/tensorflow/image_recognition/inception_v3/pruning/magnitude/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,2 @@
+intel-tensorflow==2.7.0
```
Comment (lvliang-intel) [inline examples/tensorflow/image_recognition/inception_v3/pruning/magnitude/requirements.txt:1]: use latest intel-tensorflow version?

**Review by lvliang-intel — APPROVED** [review]

**Pull request merged**